File: src/main/java/core/procedures/strings/Includes.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class Includes extends AFn {

  public Includes() {
    super(new FnArgsBuilder().min(2).max(2).mandatory(new Class[]{CharSequence.class, CharSequence.class}).build());
  }

  @Override
  public String getName() {
    return "includes?";
  }

  @Override
  public Boolean apply2(Object arg1, Object arg2) {
    return arg1.toString().contains(arg2.toString());
  }
}

File: src/main/java/core/procedures/strings/Includes.kt
package core.procedures.strings

import core.procedures.AFn
import core.procedures.Arity.Exactly

class Includes : AFn<CharSequence?, Boolean>(name = "includes?", isPure = true, arity = Exactly(2),
                                             mandatoryArgsTypes = arrayOf(CharSequence::class.java,
                                                                          CharSequence::class.java)) {

    override operator fun invoke(arg1: CharSequence?, arg2: CharSequence?) = arg1!!.contains(arg2!!)
}
